src/response.rs

```diff
@@ -0,0 +1,20 @@
+use http::{Response, Uri};
+
+use crate::body::Body;
+use crate::http;
+
+/// Extension trait for http::response::Builder objects
+///
+/// Allows the user to access the `Uri` in http::Response
+pub trait ResponseExt {
+    /// The Uri we ended up at. This can differ from the request uri when we have followed redirects.
+    fn get_uri(&self) -> &Uri;
+}
+
+impl ResponseExt for Response<Body> {
```

Comment:
Internally we always prefix some `http` crate types to make sure there is no confusion. This line should read:

`impl ResponseExt for http::Response<Body> {`